WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Amount/SimplePaymentsAmount.swift

```diff
@@ -14,33 +14,27 @@ final class SimplePaymentsAmountHostingController: UIHostingController<SimplePay
     ///
     private lazy var modalNoticePresenter: NoticePresenter = {
         let presenter = DefaultNoticePresenter()
-        presenter.presentingViewController = self
+        presenter.presentingViewController = self.navigationController ?? self
         return presenter
     }()
 
-    init(viewModel: SimplePaymentsAmountViewModel) {
+    init(viewModel: SimplePaymentsAmountViewModel, presentNoticePublisher: AnyPublisher<SimplePaymentsNotice, Never>) {
         super.init(rootView: SimplePaymentsAmount(viewModel: viewModel))
 
         // Needed because a `SwiftUI` cannot be dismissed when being presented by a UIHostingController
         rootView.dismiss = { [weak self] in
             self?.dismiss(animated: true, completion: nil)
         }
 
-        // Observe the present notice intent and set it back to `nil` after presented.
-        viewModel.$presentNotice
+        // Observe the present notice intent.
+        presentNoticePublisher
             .compactMap { $0 }
             .sink { [weak self] notice in
 
-                // To prevent keyboard to hide the notice
-                self?.view.endEditing(true)
```

Comment:
This is not needed anymore, fixed in https://github.com/woocommerce/woocommerce-ios/pull/5120